app/controllers/attributes_controller.rb

```diff
@@ -2,21 +2,25 @@ class AttributesController < ApplicationController
   include AuthenticatedApiConcern
 
   def show
-    remote_attributes = get_attributes_from_params(
+    local_attributes, remote_attributes = get_attributes_from_params(
       params.fetch(:attributes),
       permission_level: :get,
     )
 
-    render_api_response values: @govuk_account_session.get_remote_attributes(remote_attributes).compact
+    values = @govuk_account_session.get_local_attributes(local_attributes)
+      .merge(@govuk_account_session.get_remote_attributes(remote_attributes))
+
+    render_api_response values: values.compact
   end
 
   def update
-    remote_attributes = get_attributes_from_params(
+    local_attributes, remote_attributes = get_attributes_from_params(
       params.fetch(:attributes).permit!.to_h,
       permission_level: :set,
       is_hash: true,
     )
 
+    @govuk_account_session.set_local_attributes(local_attributes)
     @govuk_account_session.set_remote_attributes(remote_attributes)
 
     render_api_response
```

Comment:
Interesting, how come the render API response happens before we select/reject further down?

Reply:
I'm not sure what you mean?  The select / reject is done in a method called further up.

Reply:
Sorry I wasn't very clear in any of these feedback messages 🤦🏻 
So this one I mean that we're calling `render_api_response` does that not return at this point?

Then we're doing more filtering below that to finally return an array.

I would have thought a render statement would come at the end of a controller method, so I'm a little unsure what's going on here?